src/App.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import './App.scss';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+const preparedTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
+
+export class App extends Component {
+  state = {
+    todos: [...preparedTodos],
+    userName: '',
+    titleOfTodo: '',
+    titleError: false,
+    userError: false,
+  };
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value.trimLeft().replace(/\s{2,}/g, ' '),
```

Comment:
spaces on the right side are also possible

Reply:
fixed